lms/lib/courseware_search/lms_filter_generator.py

```diff
@@ -38,7 +42,24 @@ def filter_dictionary(self, **kwargs):
                     user,
                     cohorted_user_partition,
                 )
-                filter_dictionary['content_groups'] = unicode(partition_group.id) if partition_group else None
+
+                if partition_group:
+                    content_groups.append(unicode(partition_group.id))
+
+            course = modulestore().get_course(course_key)
+
+            for split_test_user_partition in get_split_user_partitions(course.user_partitions):
```

Comment:
This code seems to repeat (okay, almost repeat) the operation for cohorted_user_partition. Is there any way to consolidate?